app/controllers/vm_infra_controller.rb

```diff
@@ -74,15 +74,44 @@ def skip_breadcrumb?
   menu_section :inf
   has_custom_buttons
 
+  def simple_dialog_initialize(ra, options)
+    @edit = {}
+    @edit[:new] = options[:dialog] || {}
+    @edit[:wf] = ResourceActionWorkflow.new(@edit[:new], current_user, ra, {})
+    @record = Dialog.find_by(:id => ra.dialog_id.to_i)
+    @edit[:rec_id]   = @record.id
+    @edit[:key]      = "dialog_edit__#{@edit[:rec_id] || "new"}"
+    @edit[:explorer] = @explorer ? @explorer : false
```

Comment:
nitpick: can be written: 
```ruby
@edit[:explorer] = @explorer || false
```

Reply:
Plus
```
@edit = { 
  :this =>
  :that =>
....
}
```

would make more sense here rather than `@edit[:this key] = ... \n@edit[:that key] = ....

Reply:
my note tested refactor for this would be something along these lines:
```ruby

  def simple_dialog_initialize(ra, options)
    @record = Dialog.find_by(:id => ra.dialog_id.to_i)
    new? = options[:dialog] || {}
    @edit = {
      new: new?,
      wf: ResourceActionWorkflow.new(new?, current_user, ra, {}),
      rec_id: @record.id,
      key: "dialog_edit__#{@record.id || "new"}",
      explorer: @explorer : false,
      dialog_mode: options[:dialog_mode],
      current: copy_hash(new?),
      right_cell_text: options[:header].to_s
    }
    @in_a_form = true
    @changed = session[:changed] = true
    if @edit[:explorer]
      replace_right_cell(:action => 'dialog_provision')
    else
      javascript_redirect(:action => 'dialog_load')
    end
  end
```